src/libraries/System.Private.CoreLib/src/System/IO/Strategies/FileStreamHelpers.cs

```diff
@@ -35,5 +35,11 @@ private static FileStreamStrategy WrapIfDerivedType(FileStream fileStream, FileS
             e is UnauthorizedAccessException ||
             e is NotSupportedException ||
             (e is ArgumentException && !(e is ArgumentNullException));
+
+        internal static bool ShouldPreallocate(long allocationSize, FileAccess access, FileMode mode)
+            => allocationSize > 0
```

Comment:
In the PR description, you explained that you want to silently allow `allocationSize` to be a negative value, in which case, we won't allocate.

Can you elaborate some more? Why don't we throw on a negative value? I understand 0 to be silently allowed, since it means we aren't interested in allocating anything. But negative does not make sense.

Reply:
@carlossanlop based on your feedback I've changed the code to throw for negative values. So now we have:

* negative -> throw
* zero -> do nothing
* positive -> preallocate disk size